src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -1022,35 +1030,53 @@ func (c *controller) updateNodesLabels() bool {
 		return KeepWaiting
 	}
 
-	retry := false
-
+	hostsWithLabels := make(map[string]inventory_client.HostData)
```

Comment:
All you need is the labels not all the HostData
```suggestion
	hostsWithLabels := make(map[string]string)
```

Reply:
not right, i need host data in order to correlate k8s node with assisted host and hostname is not enough

Reply:
OK